### PR TITLE
#22124  Check we're not logged in from the FE before redirecting to the BE edit mode.

### LINKED ISSUE #22124: When logged in to front end, automatically redirect to back-end edit mode
**Describe the bug**
If a user has both roles (BE and FE) and logins into the FE, if you paste/type the URL it automatically redirects you to BE to the page in edit mode.

**To Reproduce**
1. Create a user with both roles (**FE and BE**) and also add the following Roles:
Screen Shot 2022-05-02 at 8 21 16 AM

2. In a private window, go to the FE and access the Members Only Section.
3. Type for example `destinations` after the host.
4. It redirects you to the destinations index page but in the BE edit mode.

**Expected behavior**
Should show the page in FE mode.

**Workaround**
Navigate to the desired page using the nav.

https://user-images.githubusercontent.com/5278970/166250724-42ef52d8-6a25-4faa-bd9c-fec425444780.mov

Customer ticket: https://dotcms.zendesk.com/agent/tickets/107078
Also reproducible in demo (22.03).

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
Check we're not logged in from the FE before redirecting to the BE edit mode.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 1]
```diff
@@ -173,4 +186,18 @@ private void goToEditPage(final String requestURI, final HttpServletRequest requ
     response.sendRedirect(url);
   }
 
+    /**
+     * This revise if the use is logged in from the frontend
+     * @param request
+     * @return
+     */
+  private boolean isFrontendLogin(final HttpServletRequest request){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:194]: very good

**Review by jdotcms — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java [hunk 2]
```diff
@@ -67,7 +80,7 @@ protected final void service(HttpServletRequest req, HttpServletResponse respons
         final VelocityRequestWrapper request =VelocityRequestWrapper.wrapVelocityRequest(req);
         final String uri = CMSUrlUtil.getCurrentURI(request);
         final boolean comeFromSomeWhere = request.getHeader("referer") != null;
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:82]: I think 'comeFromSomeWhere' is not need anymore, right?

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java [hunk 3]
```diff
@@ -357,4 +365,66 @@ public void testingTimeMachine () throws Exception{
...
+                pageContent);
+
+        final VelocityServlet velocityServlet = new VelocityServlet(WebAPILocator.getUserWebAPI(), pageAssetRenderedAPI);
+        velocityServlet.service(velocityRequest, response);
+        if(LoginMode.FE == mode){
+            //Here we verify the page was served normally
+            verify(outputStream, times(1)).write(pageContent.getBytes());
+        } else {
+            //Here the page never got served and a redirect occurred taking the user to EditMode
+            verify(outputStream, never()).write(pageContent.getBytes());
```
Comment (freddyucv) [inline dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:425]: but in this case is not called but with the page's EDIT vesion?

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/servlet/VelocityServletIntegrationTest.java:425]: when EDIT Mode is "called" or whatever it's redirect therefore the outputstream write never gets called

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityServlet.java:82]: dunno man.

**Review by nollymar — APPROVED** [review]

**Pull request merged**